modules/imap/handler_modules.php

```diff
@@ -852,6 +852,14 @@ public function process() {
                     $msg['server_id'] = $form['imap_server_id'];
                     $msg['server_name'] = $details['name'];
                     $msg['folder'] = $form['folder'];
+                    $uid = $msg['uid'];
+                    $part = true;
+                    $max = 87;
+                    $msg_struct = $imap->get_message_structure($uid);
+                    $struct = $imap->search_bodystructure($msg_struct, array('imap_part_number' => $part));
+                    $msg_struct_current = array_shift($struct);
+                    $msg_text = $imap->get_message_content($uid, $part, $max, $msg_struct_current);
+                    $msg['preview_msg'] = $msg_text;
```

Comment:
The way you do it has several problems:
1. Too many internal calls to imap method.
2. Too many calls to imap server for each message - it will be rather slow for bigger lists of messages. Might slow down the normal list pages by 2-4 times.
3. I think we should seriously do this conditionally based on a setting that user turns on and off for specific pages (imap folder pages, combined pages).
4. The code to get the preview should be moved out of here as this only works for single imap pages. We have many other places listing messages (search results, combined pages - all, sent, unread, etc.)
5. It is best to have a setting, pass it to the imap get_mailbox_page or get_message_list methods and have it query the text body as well if required. This will ensure we get all the content in a single call to the imap server. Please test this with 50 messages at once and you will notice the difference. Check if the IMAP protocol doesn't have something built-in - like getting part of the text body of the message or getting only the text body, not the whole structure. Parsing the bodystructure is an expensive operation. If we really need to do this for each message separately in list-view, we should consider aggressive caching.

Reply:
Noted